pkg/apis/monitoring/v1/types.go

```diff
@@ -1010,6 +1016,12 @@ func (c *TLSConfig) Validate() error {
 		return &TLSConfigValidationError{"tls config can not both specify KeyFile and KeySecret"}
 	}
 
+	if c.CertFile != "" && (c.KeyFile == "" || c.KeySecret == nil) {
+		return &TLSConfigValidationError{"client cert specified without client key"}
+	} else if c.KeyFile != "" && (c.CertFile == "" || c.Cert == (SecretOrConfigMap{})) {
```

Comment:
stylistic nit: the `else if` is not necessary. it is more readable if you simply add a newline with a new `if` statement.

Reply:
concretely:
```
	if c.Cert != (SecretOrConfigMap{}) && c.KeySecret == nil {
		return &TLSConfigValidationError{"client cert specified without client key"}
	}

	if c.KeySecret != nil && c.Cert == (SecretOrConfigMap{}) {
		return &TLSConfigValidationError{"client key specified without client cert"}
	}

	return nil
```